packages/core/src/shared/logger/toolkitLogger.ts

```diff
@@ -156,7 +156,7 @@ export class ToolkitLogger extends BaseLogger implements vscode.Disposable {
 
         meta = meta.map((o) => (o instanceof Error ? this.mapError(level, o) : o))
 
-        if (messageWithHeader instanceof Error) {
+        if (typeof messageWithHeader === 'object' && messageWithHeader !== null) {
```

Comment:
Checking object in this case is a bit of a hack so we want to avoid this when possible. My comment above should allow us to revert this.

Reply:
I'm not sure if I completely understand your solution. But I agree with avoiding this check.

What do you think about declaring a type:
`type LogMessage = { type: 'string'; content: string } | { type: 'error'; topic: string; error: Error }
`?

So that the `addTopicToMessage` would return a `LogMessage`, and for this check we can have a switch of:
`switch (logMessage.type) {
        case 'string':
            this.logger.log(level, logMessage.content, ...meta, { logID: this.idCounter })
            break
        case 'error':
            this.logger.log(level, '%O', { topic: logMessage.topic, message: logMessage.error }, ...meta, { logID: this.idCounter })
            break
    }`